### ISSUE TITLE
Add compatibility with Symfony 7

### ISSUE THREAD

**Issue by treeindark** [ratchetphp/Ratchet#1045]
Hello, did you plan to upgrade ratchet, for use with symfony 7 ?  -- deprecated to remove -- want some help ?

**Comment by treeindark** [comment]
https://github.com/ratchetphp/Ratchet/pull/1053

It work, with fix (send patch-1) @joem236

**Comment by SimonFrings** [comment]
Hey @treeindark, I agree that we should make Ratchet compatible with Symfony 7. The pull request in #1053 is a good start for this, but there are a lot of changes that involve BC breaks and should probably be addressed in separate pull requests. Changing the range of supported PHP versions will probably be addressed in the future, so for now I think we have to make this work across our current PHP version range.

We're currently looking into project funding (as discussed in #1054) and once we know how to proceed with Ratchet, we first want to add support for the latest PHP and PHPUnit versions. We'll look into additional compatibility PRs (like Symfony 7) afterwards. I will keep this ticket updated as we make progress with this :+1:

**Comment by Tofandel** [comment]
I agree, that PR, while great for a new version contains breaking changes, not necessary to support v7, in fact just adding

```
      "symfony/http-foundation": "^2.6|^3.0|^4.0|^5.0|^6.0|^v7.0",
      "symfony/routing": "^2.6|^3.0|^4.0|^5.0|^6.0|^7.0",
```

In composer.json is enough as there doesn't seem to be anything in the PR that addresses a breaking change from symfony

**Comment by clue** [comment]
@treeindark Thanks for reporting and for your patience, I know it's been a while! :)

I've just merged #1098 which should finally bring full support for Symfony 7. This is part of a larger effort reviving Ratchet as discussed in #1054, solving many other issues along the way.

If you'd like to support us on this journey, consider [sponsoring the project](https://github.com/sponsors/reactphp). ❤️
